tests/test_cached_files.py

```diff
@@ -8,21 +8,96 @@
 from cashctrl_api import CachedCashCtrlClient, CashCtrlClient
 
 @pytest.fixture(scope="module")
-def cc_client():
-    return CachedCashCtrlClient()
+def tmp_path_for_module(tmp_path_factory):
+    return tmp_path_factory.mktemp("temp")
 
 @pytest.fixture(scope="module")
-def files():
+def mock_directory(tmp_path_for_module):
+    """Create a temporary directory, populate with files and folders."""
+    tmp_path = tmp_path_for_module
+    (tmp_path / 'file1.txt').write_text("This is a text file.")
+    (tmp_path / 'file2.log').write_text("Log content here.")
+    subdir = tmp_path / 'subdir'
+    subdir.mkdir(exist_ok=True)
+    (subdir / 'file3.txt').write_text("A Text file in a subdirectory.")
+    nested_subdir = tmp_path / 'nested' / 'subdir'
+    nested_subdir.mkdir(parents=True, exist_ok=True)
+    (nested_subdir / 'file4.txt').write_text("File in the nested directory.")
+    (nested_subdir / 'file5.log').write_text("Another nested directory file.")
+    return tmp_path
+
+@pytest.fixture(scope="module")
+def cc_client(mock_directory):
+    """Create a CachedCashCtrlClient, populate with files and folders."""
     cc_client = CachedCashCtrlClient()
+    initial_files = cc_client.list_files()
+    # check is this needed
+    cc_client.mirror_directory(mock_directory, delete_files=False)
+
+    # We create a fresh instance with empty cache, because the cache is
+    # populated when mirroring a directory
+    cc_client = CachedCashCtrlClient()
+
+    yield cc_client
+
+    files = cc_client.list_files()
+    # check is this needed
+    to_delete = set(files['id']).difference(initial_files['id'])
+    params = { 'ids': ','.join(str(i) for i in to_delete), 'force': True }
+    cc_client.post("file/delete.json", params=params)
+    # Empty recycle bin to release references before category deletion
```

Comment:
Let's be less invasive and leave the recycle bin untouched. It seems that just deleting it is not sufficient to delete file categories.